django/conf/__init__.py

```diff
@@ -140,7 +140,14 @@ def __init__(self, settings_module):
         # store the settings module in case someone later cares
         self.SETTINGS_MODULE = settings_module
 
-        mod = importlib.import_module(self.SETTINGS_MODULE)
+        try:
+            mod = importlib.import_module(self.SETTINGS_MODULE)
+        except ImportError as exc:
+            # If the settings module cannot be imported, treat it as a configuration error.
+            # Inability to import any other module should not be treated as a configuration error.
+            if self.SETTINGS_MODULE in exc.msg:
```

Comment:
Can you detect this in a more structured way, e.g. by using the `name` or `path` attributes? https://docs.python.org/3/library/exceptions.html#ImportError

Reply:
Thanks for pointing that out, I've updated the PR.